c7n/filters/offhours.py

```diff
@@ -495,9 +497,10 @@ def parse(self, tag_value):
         return schedule
 
     @staticmethod
-    def has_resource_schedule(tag_value):
+    def has_resource_schedule(tag_value, time_type):
         raw_data = ScheduleParser.raw_data(tag_value)
-        return 'on' in raw_data and 'off' in raw_data
+        # note time_type is set to 'on' or 'off' and raw_data is a dict
```

Comment:
if it's an onhours policy that is running, 'on' is required to be in raw_data. if it's an offhours policy that is running 'off' is required to be in raw_data